File: RPG City Lite Server/src/main/java/com/jbcomputers/rpgcitylite/entities/BuildingRequirement.java
package com.jbcomputers.rpgcitylite.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@Entity
@Table(name = "building_reqs")
public class BuildingRequirement {

    @Id
    @GeneratedValue(strategy =  GenerationType.AUTO)
    @Column(name="reqs_id")
    Long reqsId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name="buildingid", nullable = false)
    @JsonBackReference
    Building building;

    @Column(name="category")
    String category;

    @Column(name="type")
    String type;

    @Column(name="value")
    int value;

    @Column(name="level")
    int level;

    @Column(name="image")
    String image;
}

File: RPG City Lite Client/RPG City Lite/Assets/Scripts/Entities/BuildingRequirement.cs

public class BuildingRequirement 
{
    long reqsId;
    string category;
    string type;
    int value;
    int level;
    string image;

    public long ReqsId { get => reqsId; set => reqsId = value; }
    public string Category { get => category; set => category = value; }
    public string Type { get => type; set => type = value; }
    public int Value { get => value; set => this.value = value; }
    public int Level { get => level; set => level = value; }
    public string Image { get => image; set => image = value; }
}
